File: csharp/src/ev1/ut2_controlflujometodos/u04metodos/ejercicios/pendientes/E02_SumaDos_Pendiente.cs
/*
OBJETIVO: Metodo int Sumar(int a, int b); imprime Sumar(4,6).
PATRON: Menu do-while con opcion 0 salir y uso de Console.ReadLine().
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/
using System;
class Program
{
    static void Main()
    {
        string? opcion;
        do
        {
            Console.WriteLine();
            Console.WriteLine("=== E02 Suma Dos Pendiente ===");
            Console.WriteLine("1. Preparar y revisar el ejercicio");
            Console.WriteLine("0. Salir");
            Console.Write("Elige una opcion: ");
            opcion = Console.ReadLine();
            switch (opcion)
            {
                case "1":
                    Console.WriteLine("Completa el metodo Sumar(int a, int b).");
                    Console.WriteLine("Prueba preparada: Sumar(4, 6).");
                    Console.WriteLine($"Resultado actual del esqueleto: {Sumar(4, 6)}");
                    break;
                case "0":
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opcion no valida.");
                    break;
            }
        } while (opcion != "0");
    }
    static int Sumar(int a, int b)
    {
        // TODO: devolver la suma de a y b.
        return 0;
    }
}

File: cpp/src/ev1/ut2_controlflujometodos/u04metodos/ejercicios/pendientes/E02_SumaDos_Pendiente.cpp
/*
OBJETIVO: Funcion int Sumar(int a, int b); imprime Sumar(4,6).
PATRON: Menu do-while con opcion 0 salir y uso de cin.
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

void ImprimirMenu() {
    cout << endl;
    cout << "=== E02 Suma Dos Pendiente ===" << endl;
    cout << "1. Preparar y revisar el ejercicio" << endl;
    cout << "0. Salir" << endl;
}

int Sumar(int a, int b)
{
    // TODO: devolver la suma de a y b.
    return 0;
}

int main() {
    int opcion;
    do {
        ImprimirMenu();
        cout << "Introduce una opcion -> ";
        cin >> opcion;
        switch (opcion) {
            case 1:
                cout << "Completa la funcion Sumar(int a, int b)." << endl;
                cout << "Prueba preparada: Sumar(4, 6)." << endl;
                cout << "Resultado actual del esqueleto: " << Sumar(4, 6) << endl;
                break;
            case 0:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion no valida." << endl;
                break;
        }
    } while (opcion != 0);
    return 0;
}
